src/main/java/com/example/enterpriseapplication/AddRecipeController.java

```diff
@@ -26,22 +26,14 @@ public String addRecipe(@RequestParam("recipeName") String recipeName,
                             @RequestParam("description") String description) throws Exception {
         Recipe recipe = new Recipe();
         recipe.setRecipeName(recipeName);
-        recipe.setRecipeIngredients(parseIngredients(ingredients));
-        recipe.setRecipeSteps(parseSteps(description));
+        recipe.setRecipeIngredients(ingredients.split("\\r?\\n"));
+        recipe.setRecipeSteps(description.split("\\r?\\n"));
```

Comment:
# Change 1

I put the logic for parsing the input values from the request in the endpoint method. This was done because keeping these methods in the same class makes the class have mixed purposes. Keeping the purpose of the class consistent helps with keeping the code clean.

Keeping the class orgnaized this way can help with the first part of the SOLID principles: https://medium.com/@edin.sahbaz/comprehensive-guide-to-solid-principles-in-c-54d79e19b7d7

An alternative to this change is to change this data to go through the body (which in the long run, if you want to send more information than this I recommend). I would also recommend going through the body if the description gets long.

When sending values through the query-string, there is a maximum character count which you could reach if the user sends a large amount of steps.